### PR TITLE
[xdoctest][task 304] reformat example code with google style in python/paddle/incubate/tensor/math.py

### PR DESCRIPTION
**Pull request by yoyoIcy**
### PR types

Others

### PR changes

Others

### Description

修改如下文件的示例代码，使其通过 `xdoctest` 检查：

- `python/paddle/incubate/tensor/math.py`

预览：

- en: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/jit/XXX_en.html
- zh: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/jit/XXX_cn.html

### Related links
 
- #55629
- #55295

@sunzhongkai588 @SigureMo @megemini

### CODE REVIEW

File: python/paddle/incubate/tensor/math.py [hunk 1]
```diff
@@ -56,11 +56,14 @@ def segment_sum(data, segment_ids, name=None):
...
-            out = paddle.incubate.segment_sum(data, segment_ids)
-            #Outputs: [[4., 4., 4.], [4., 5., 6.]]
+            >>> import paddle
+            >>> data = paddle.to_tensor([[1, 2, 3], [3, 2, 1], [4, 5, 6]], dtype='float32')
+            >>> segment_ids = paddle.to_tensor([0, 0, 1], dtype='int32')
+            >>> out = paddle.incubate.segment_sum(data, segment_ids)
+            >>> print(out)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[4., 4., 4.],
+            [4., 5., 6.]])
```
Comment (SigureMo) [inline python/paddle/incubate/tensor/math.py]:
```suggestion
            [[4., 4., 4.],
             [4., 5., 6.]])
```

File: python/paddle/incubate/tensor/math.py [hunk 2]
```diff
@@ -121,11 +124,14 @@ def segment_mean(data, segment_ids, name=None):
...
-            out = paddle.incubate.segment_mean(data, segment_ids)
-            #Outputs: [[2., 2., 2.], [4., 5., 6.]]
+            >>> import paddle
+            >>> data = paddle.to_tensor([[1, 2, 3], [3, 2, 1], [4, 5, 6]], dtype='float32')
+            >>> segment_ids = paddle.to_tensor([0, 0, 1], dtype='int32')
+            >>> out = paddle.incubate.segment_mean(data, segment_ids)
+            >>> print(out)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[2., 2., 2.],
+            [4., 5., 6.]])
```
Comment (SigureMo) [inline python/paddle/incubate/tensor/math.py]:
```suggestion
            [[2., 2., 2.],
             [4., 5., 6.]])
```

File: python/paddle/incubate/tensor/math.py [hunk 3]
```diff
@@ -186,11 +192,14 @@ def segment_min(data, segment_ids, name=None):
...
-            out = paddle.incubate.segment_min(data, segment_ids)
-            #Outputs:  [[1., 2., 1.], [4., 5., 6.]]
+            >>> import paddle
+            >>> data = paddle.to_tensor([[1, 2, 3], [3, 2, 1], [4, 5, 6]], dtype='float32')
+            >>> segment_ids = paddle.to_tensor([0, 0, 1], dtype='int32')
+            >>> out = paddle.incubate.segment_min(data, segment_ids)
+            >>> print(out)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[1., 2., 1.],
+            [4., 5., 6.]])
```
Comment (SigureMo) [inline python/paddle/incubate/tensor/math.py]:
```suggestion
            [[1., 2., 1.],
             [4., 5., 6.]])
```

File: python/paddle/incubate/tensor/math.py [hunk 4]
```diff
@@ -251,11 +260,14 @@ def segment_max(data, segment_ids, name=None):
...
-            out = paddle.incubate.segment_max(data, segment_ids)
-            #Outputs: [[3., 2., 3.], [4., 5., 6.]]
+            >>> import paddle
+            >>> data = paddle.to_tensor([[1, 2, 3], [3, 2, 1], [4, 5, 6]], dtype='float32')
+            >>> segment_ids = paddle.to_tensor([0, 0, 1], dtype='int32')
+            >>> out = paddle.incubate.segment_max(data, segment_ids)
+            >>> print(out)
+            Tensor(shape=[2, 3], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [[3., 2., 3.],
+            [4., 5., 6.]])
```
Comment (SigureMo) [inline python/paddle/incubate/tensor/math.py]:
```suggestion
            [[3., 2., 3.],
             [4., 5., 6.]])
```

**Review by megemini — APPROVED** [review]
LGTM

**Review by SigureMo — APPROVED** [review]
LGTMeow 🐾

**Pull request merged**